marlowe-actus/src/Language/Marlowe/ACTUS/Model/APPLICABILITY/Applicability.hs

```diff
@@ -3,72 +3,125 @@ module Language.Marlowe.ACTUS.Model.APPLICABILITY.Applicability where
 
 import           Data.Maybe                                                    (isJust)
 import           Data.Validation
-import           Language.Marlowe.ACTUS.Definitions.ContractTerms              (ContractTerms (..), ContractType (..),
+import           Language.Marlowe.ACTUS.Definitions.ContractTerms              (CT (..), ContractTerms (..),
                                                                                 ScheduleConfig (..),
                                                                                 TermValidationError (..))
 import           Language.Marlowe.ACTUS.Model.APPLICABILITY.ApplicabilityModel
 
 validateTerms :: ContractTerms -> Validation [TermValidationError] ContractTerms
 validateTerms t =
     case contractType t of
-        Just PAM ->
+        PAM ->
             pure t <*
+            _NN ct_IED t "initial exchange date" <*
+            _NN ct_DCC t "day count convention" <*
             _X (calendar . scfg) t "calendar" <*
             _X (bdc . scfg) t "business day convention" <*
             _X (eomc . scfg) t "end of month convention" <*
-            -- TODO: to be used when contract term optionality is fixed
-            -- _X_I_2 [isJust $ ct_FEANX t, isJust $ ct_FECL t] t ["cycle anchor date of fee", "cycle of fee"] <*
+            _X_I_2 (ct_FER t) [isJust $ ct_FEANX t, isJust $ ct_FECL t] t "fee rate" ["cycle anchor date of fee", "cycle of fee"] <*
             _X ct_FEAC t "fee accrued" <*
+            _NN_I_1 [isJust $ ct_FEB t, isJust $ ct_FER t] t ["fee basis", "fee rate"] <*
             _NN ct_IPNR t "nominal interest rate" <*
             _X ct_IPAC t "accrued interest" <*
             _X ct_IPCED t "capitalization end date" <*
             _X ct_PDIED t "premium discount at IED" <*
-            -- TODO: to be used when contract term optionality is fixed
-            -- _X_I_2 [isJust $ ct_SCANX t, isJust $ ct_SCCL t] t ["cycle anchor date of scaling index", "cycle of scaling index"] <*
-            _X_I_1 [isJust $ ct_RRCL t, isJust $ ct_RRANX t] ct_RRNXT t ["cycle anchor date of rate reset", "cycle of rate reset"] "next reset rate" <*
+            _X_I_2 (ct_SCEF t) [isJust $ ct_SCANX t, isJust $ ct_SCCL t] t "scaling effect" ["cycle anchor date of scaling index", "cycle of scaling index"] <*
+            _X_I_1 [isJust $ ct_RRCL t, isJust $ ct_RRANX t]
+              [ isJust $ ct_RRNXT t,
+                isJust $ ct_RRSP t,
+                isJust $ ct_RRMLT t,
+                isJust $ ct_RRPF t,
+                isJust $ ct_RRPC t,
+                isJust $ ct_RRLC t,
+                isJust $ ct_RRLF t] t ["cycle anchor date of rate reset", "cycle of rate reset"]
```

Comment:
Could write `map isJust [ct_RRNXT t, ct_RRSP t, ... ]` instead

Reply:
In fact `map (\x -> isJust $ x t) [ct_RRNXT, ct_RRSP, ... ]` or a list comprehension would be more readable